packages/wallets/src/features/cashier/modules/Transfer/components/TransferFormAmountInput/TransferFormAmountInput.tsx

```diff
@@ -154,9 +163,8 @@ const TransferFormAmountInput: React.FC<TProps> = ({ fieldName }) => {
                 label={amountLabel}
                 locale={preferredLanguage}
                 maxDigits={maxDigits}
-                onBlur={() => setFieldValue('activeAmountFieldName', undefined)}
```

Comment:
This is the most important change here, i.e. we won't keep a record of what field is currently being edited, but instead which one is the most recently used.

This is because `activeAmountFieldName` is `undefined` when both fields are not in focus, in which case both `onChangeHandler` and `amountConverterHandler` don't get executed (see check in the start of each function code). These functions, in turn, are being called right after the debounce is done (500ms after the edit) and they update the non-edited field. So if the user exits the field they've just edited in this 500ms window, no functions are called and the other field remains unchanged.

Reply:
Have changed the code, the comment is no longer relevant